src/mds/MDSRank.cc

```diff
@@ -2497,6 +2499,42 @@ bool MDSRank::command_dirfrag_ls(
   return true;
 }
 
+void MDSRank::command_openfiles_ls(Formatter *f) 
+{
+  Mutex::Locker l(mds_lock);
+  
+  for (auto p : mdlog->get_segments()) {
```

Comment:
My feeling on this is that it shouldn't be necessary to traverse the log to get the inodes that are "open" (i.e. some client has caps). Unfortunately, traversing the session caps is expensive because you need to remember which inodes you've seen already. Traversing the entire `mdcache->inode_map` and `mdcache->snap_inode_map` is another option.

@ukernel what do you think?

Reply:
sorry batrick,  but i think traverse inode_map should be more expensive than segments , right ?

Reply:
Yes, but I'm concerned about the technical debt of traversing the log in `MDSRank.cc`.

Reply:
I think it's better to traverse inode_map

Reply:
understood